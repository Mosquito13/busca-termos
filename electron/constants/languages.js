module.exports = {
  USA: 'en-us',
  BRAZIL: 'pt-br',
  SPAIN: 'es',
  PORTUGAL: 'pt',
  GERMANY: 'de',
  FRANCE: 'fr',
  ROMANIA: 'ro',
  ITALIA: 'it',
  SLOVAKIA: 'sk',
  TURKEY: 'tr',
  RUSSIA: 'ru',
  POLAND: 'pl',
  CATALONIA: 'ca',
  DENMARK: 'da',
  FINLAND: 'fi',
  CZECH: 'cs'
};
