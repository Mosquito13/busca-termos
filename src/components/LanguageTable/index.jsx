import { memo, useMemo } from 'react';
import { useSelector } from 'react-redux';
import Table, { SMALL, AUTO } from '../common/Table';

import settingsSelectors from '../../selectors/settings';
import coreSelectors from '../../selectors/core';
import searchUtils from '../../utils/searchUtils';

const columnsConfiguration = [
  {
    title: 'CÃ³digo',
    dataKey: 'id',
    size: SMALL
  },
  {
    title: 'Termo',
    dataKey: 'content',
    size: AUTO
  }
];

const LanguageTable = () => {
  const data = useSelector(coreSelectors.getData);
  const filter = useSelector(coreSelectors.getFilter);
  const loading = useSelector(coreSelectors.isLoading);
  const mainLanguage = useSelector(settingsSelectors.getMainLanguage);

  const filteredData = useMemo(() => {
    if (loading) return [];

    return searchUtils.applyFilter(data, filter, mainLanguage);
  }, [loading, data, filter, mainLanguage]);

  return (
    <Table idField="id" data={filteredData} columns={columnsConfiguration} />
  );
};

export default memo(LanguageTable);
