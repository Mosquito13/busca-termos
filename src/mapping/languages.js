import { ReactComponent as BrazilIcon } from '../assets/svg/brazil.svg';
import { ReactComponent as CataloniaIcon } from '../assets/svg/catalonia.svg';
import { ReactComponent as CzechIcon } from '../assets/svg/czech.svg';
import { ReactComponent as DenmarkIcon } from '../assets/svg/denmark.svg';
import { ReactComponent as FinlandIcon } from '../assets/svg/finland.svg';
import { ReactComponent as FranceIcon } from '../assets/svg/france.svg';
import { ReactComponent as GermanyIcon } from '../assets/svg/germany.svg';
import { ReactComponent as ItalyIcon } from '../assets/svg/italy.svg';
import { ReactComponent as PolandIcon } from '../assets/svg/poland.svg';
import { ReactComponent as PortugalIcon } from '../assets/svg/portugal.svg';
import { ReactComponent as RomaniaIcon } from '../assets/svg/romania.svg';
import { ReactComponent as RussiaIcon } from '../assets/svg/russia.svg';
import { ReactComponent as SlovakiaIcon } from '../assets/svg/slovakia.svg';
import { ReactComponent as SpainIcon } from '../assets/svg/spain.svg';
import { ReactComponent as TurkeyIcon } from '../assets/svg/turkey.svg';
import { ReactComponent as UsaIcon } from '../assets/svg/usa.svg';

const languageMapping = {
  BRAZIL: {
    id: 'pt-br',
    title: 'Português (Brasil)',
    getIcon: () => <BrazilIcon />
  },
  CATALONIA: {
    id: 'ca',
    title: 'Catalão',
    getIcon: () => <CataloniaIcon />
  },
  CZECH: {
    id: 'cs',
    title: 'Tcheco',
    getIcon: () => <CzechIcon />
  },
  DENMARK: {
    id: 'da',
    title: 'Dinamarquês',
    getIcon: () => <DenmarkIcon />
  },
  FINLAND: {
    id: 'fi',
    title: 'Finlandês',
    getIcon: () => <FinlandIcon />
  },
  FRANCE: {
    id: 'fr',
    title: 'Francês',
    getIcon: () => <FranceIcon />
  },
  GERMANY: {
    id: 'de',
    title: 'Alemão',
    getIcon: () => <GermanyIcon />
  },
  ITALY: {
    id: 'it',
    title: 'Italiano',
    getIcon: () => <ItalyIcon />
  },
  POLAND: {
    id: 'pl',
    title: 'Polonês',
    getIcon: () => <PolandIcon />
  },
  PORTUGAL: {
    id: 'pt',
    title: 'Português (Portugal)',
    getIcon: () => <PortugalIcon />
  },
  ROMANIA: {
    id: 'ro',
    title: 'Romeno',
    getIcon: () => <RomaniaIcon />
  },
  RUSSIA: {
    id: 'ru',
    title: 'Russo',
    getIcon: () => <RussiaIcon />
  },
  SLOVAKIA: {
    id: 'sk',
    title: 'Eslovaco',
    getIcon: () => <SlovakiaIcon />
  },
  SPAIN: {
    id: 'es',
    title: 'Espanhol',
    getIcon: () => <SpainIcon />
  },
  TURKEY: {
    id: 'tr',
    title: 'Turco',
    getIcon: () => <TurkeyIcon />
  },
  USA: {
    id: 'en-us',
    title: 'Inglês (EUA)',
    getIcon: () => <UsaIcon />
  }
};

export { languageMapping };
