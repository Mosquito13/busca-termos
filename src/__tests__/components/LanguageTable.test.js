import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';

import LanguageTable from '../../components/LanguageTable';
import { languageMapping } from '../../mapping/languages';

const dataMock = {
  [languageMapping.USA.id]: [
    {
      uid: 1,
      id: '100000',
      content: 'Incluir'
    }
  ]
};

const getStateMock = (loading) => ({
  Core: {
    loading,
    data: dataMock,
    filter: ''
  },
  Settings: {
    mainLanguage: languageMapping.USA.id
  }
});

describe(`${LanguageTable.type.name}`, () => {
  describe('snapshots', () => {
    beforeAll(() => {
      jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => jest.fn());
    });

    it('should render loading', () => {
      jest
        .spyOn(reactRedux, 'useSelector')
        .mockImplementation((selector) => selector(getStateMock(true)));

      const tree = renderer.create(<LanguageTable />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render without loading', () => {
      jest
        .spyOn(reactRedux, 'useSelector')
        .mockImplementation((selector) => selector(getStateMock(false)));

      const tree = renderer.create(<LanguageTable />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
