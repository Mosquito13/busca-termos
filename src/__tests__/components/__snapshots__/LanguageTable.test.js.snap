// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageTable snapshots should render loading 1`] = `
<div
  className="table"
>
  <div
    className="table__header"
  >
    <div
      className="table__row table__row--header"
    >
      <div
        className="table__cell table__cell--header table__cell--small"
        onClick={[Function]}
      >
        <span
          className="table__cell-text"
        >
          Código
        </span>
      </div>
      <div
        className="table__cell table__cell--header"
        onClick={[Function]}
      >
        <span
          className="table__cell-text"
        >
          Termo
        </span>
      </div>
    </div>
  </div>
  <div
    className="table__body"
  >
    <div
      className="table__empty-state"
    >
      <div
        className="table__empty-state-message"
      >
        <h4
          className="heading-4"
        >
          Nenhum resultado encontrado.
        </h4>
      </div>
      <div
        className="table__empty-state-image-wrapper"
      >
        <div
          className="table__empty-state-image"
        >
          <svg>
            emptystate.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LanguageTable snapshots should render without loading 1`] = `
<div
  className="table"
>
  <div
    className="table__header"
  >
    <div
      className="table__row table__row--header"
    >
      <div
        className="table__cell table__cell--header table__cell--small"
        onClick={[Function]}
      >
        <span
          className="table__cell-text"
        >
          Código
        </span>
      </div>
      <div
        className="table__cell table__cell--header"
        onClick={[Function]}
      >
        <span
          className="table__cell-text"
        >
          Termo
        </span>
      </div>
    </div>
  </div>
  <div
    className="table__body"
  >
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
